gramineproject/gramine
# [DRAFT] [WIP] [DONOTMERGE] [python] Add support for production signing with Azure Key Vault

Signed-off-by: Sankaranarayanan Venkatasubramanian <sankaranarayanan.venkatasubramanian@intel.com>



## Description of the changes 
This change enables production signing with Azure Key Vault's Managed HSM.


#11 
#698 

## How to test this PR? 

- You need to have Azure Subscription with access to Azure Key Vault's Managed HSM
- Create a 3072-bit RSA key in Azure Key Vault's managed HSM with public exponent of 3 (Please note that only Managed HSM supports setting public exponent), and give permissions to yourself or who has to use to the key created as 'Managed HSM Crypto User' (az keyvault role assignment).
- Set environment variable VAULT_URL to your Managed HSM's URL (typically of the form https://<your-managed-hsm-name.managedhsm.azure.net).
- gramine-sgx-sign is modified to take a keytype parameter that will use pem for local private key and akv for Azure Key Vault
- It also has a change to interpret key as file path if it keytype is pem and as key name (created in AKV) if keytype is AKV.
- Default options for gramine-sgx-sign does not change (keytype is pem and key is SGX_RSA_KEY_PATH).
- Please see helloworld Makefile for the change (I will remove it in the final PR and create a separate on in gramineproject/examples).
- I will expand the documentation in the final PR (currently enabling the gsc part for the same)

This was tested with my Azure Key Vault subscription, and @dimakuv onscreen reviewed it as part of execution (Thanks, @dimakuv)
